internals-js/src/__tests__/subgraphValidation.test.ts

```diff
@@ -861,4 +861,20 @@ describe('@core/@link handling', () => {
       '[S] Invalid definition for directive "@key": argument "fields" should have type "federation__FieldSet!" but found type "federation__FieldSet"',
     ]]);
   });
+
+  it('allows any (non-scalar) type in redefinition when expected type is a scalar', () => {
+    const doc = gql`
+      extend schema
+        @link(url: "https://specs.apollo.dev/federation/v2.0", import: ["@key"])
+
+      type T @key(fields: "k") {
+        k: ID!
+      }
+
+      directive @key(fields: String!, resolvable: Boolean = true) repeatable on OBJECT | INTERFACE
```

Comment:
**observation** It took me a sec to realize we were testing that `fields: String!` was allowed instead of

```graphql
scalar _FieldSet
directive @key(fields: _FieldSet!, resolveable: Boolean = true) repeatable on OBJECT | INTERFACE
```

I assume eventually I'll be able to recite the core schema spec by memory

Reply:
Added a comment so no-one else has to waste time on it :).